Wrappers/Python/cil/optimisation/algorithms/SPDHG.py

```diff
@@ -42,202 +46,460 @@ class SPDHG(Algorithm):
     operator : BlockOperator
         BlockOperator must contain Linear Operators
     tau : positive float, optional, default=None
-        Step size parameter for Primal problem
+        Step size parameter for primal problem. If `None` see note.
     sigma : list of positive float, optional, default=None
-        List of Step size parameters for Dual problem
-    initial : DataContainer, optional, default=None
+        List of Step size parameters for dual problem. If `None` see note.
+    initial : DataContainer, optional, the default value is a zero DataContainer in the range of the `operator`.
         Initial point for the SPDHG algorithm
-    prob : list of floats, optional, default=None
-        List of probabilities. If None each subset will have probability = 1/number of subsets
-    gamma : float
-        parameter controlling the trade-off between the primal and dual step sizes
+    gamma : float, optional
+            Parameter controlling the trade-off between the primal and dual step sizes
+    sampler: optional, an instance of a `cil.optimisation.utilities.Sampler` class or another class with the function __next__(self) implemented outputting an integer from {1,...,len(operator)}. 
```

Comment:
Is it really from `{1,...,len(operator)}`?

We should use zero indexing everywhere so this might be a typo or a bigger issue.

Reply:
Good spot, we do index from 0 so a typo and not a bigger issue